doc/flame/components.md

```diff
@@ -154,6 +154,48 @@ void update(double dt) {
 ```
 
 
+### Blueprints
+
+Flame also offers a virtual approach for grouping components which allows the developer to keep
+their components grouped "code wise", but the components (or other things, like a
+`ContactCallback` for Forge2d Games) are individually added to the game and there isn't a parent
+component.
+
+This is useful when you have a complex feature that will require different components, but they
+all need to be on the same rendering level of different faatures, a simple example for this is
+a top down view game, we would like to play with the rendering order of things so we can give the
+player the impression of depth, so if the player character goes behind a house, its roofs is
+rendered "above" the player, to be able to achieve that we would need to split our building
+component into more components, for example: `HouseBase`, `HouseSecondFloor` and `HouseRoof`.
```

Comment:
```suggestion
This is useful when you have several multi-part components that each need to control the rendering level of its parts globally. A simple example would be a top-down game, in which in order to give the impression of depth, when player character goes behind a house, its roof is rendered "above" the player. In this case, you could still "group" all the building components (`HouseBase`, `HouseSecondFloor`, `HouseRoof`) with a Blueprint instead of a parent component, while they are still all siblings to the Player component.
```